src/main/java/TaskList.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Scanner;
+public class TaskList {
+        private final Task[] taskList;
+        private static final int MAX_TASKS = 100;
+        //protected boolean[] isMarked;
+        private int taskCount;
+
+
+        public TaskList() {
+            this.taskList = new Task[MAX_TASKS];
+            this.taskCount = 0;
+        }
+
+    private void processTaskCommand(String command) {
+        String[] commandParts = command.split(" ", 2);
+        String taskType = commandParts[0];
+
+        switch (taskType) {
+        case "todo":
+            addTask(new Task(commandParts[1]));
+            break;
+        case "deadline":
+            String[] deadlineParts = commandParts[1].split("/by" , 2);
+            if (deadlineParts.length != 2) {
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                addTask(new Deadline(deadlineParts[0], deadlineParts[1]));
+            }
+            break;
+        case "mark":
+            if (commandParts.length == 1 || !isValidIndex(Integer.parseInt(commandParts[1])-1) ) {
+                System.out.println("Invalid command format for marking a task. Use 'mark <index>'.");
+            } else {
+                markTask(Integer.parseInt(commandParts[1]) - 1);
+            }
+            break;
+        case "unmark":
+            if (commandParts.length == 1 || !isValidIndex(Integer.parseInt(commandParts[1])-1) ) {
+                System.out.println("Invalid command format for marking a task. Use 'mark <index>'.");
+            } else {
+                unmarkTask(Integer.parseInt(commandParts[1]) - 1);
+            }
+            break;
+        default:
+            System.out.println("☹ Dobby does not understand."); // Default to addTask if not recognised
+            break;
+        }
+    }
+
+        /**
+         * Adds user input into taskList
+         *
+         * @param task input given by user
+         */
+        public void addTask(Task task) {
+            taskList[taskCount] = task;
+            //isMarked[taskCount] = false;
```

Comment:
you should consider deleting out commented code to not confuse the person reading it